docs/tutorials/data_management.md

```diff
@@ -0,0 +1,243 @@
+
+Let's give it a try:
+
+```console
+$ icu4x-datagen --cldr-tag latest --icuexport-tag latest --out my-data --format mod --keys-for-bin target/debug/myapp --locales ja
+```
+
+The output might tell you additional crates that need to be installed. Don't worry, these are transitive dependencies already anyway, but are required directly now to construct our data:
+
+```console
+$ cargo add icu_provider
+$ cargo add litemap
+$ cargo add zerovec
+```
+
+We can then use the data by directly including the source with the `include!` macro.
+
+```rust
+extern crate alloc; // required as BakedDataProvider is written for #[no_std]
+use icu::locid::{locale, Locale};
+use icu::calendar::DateTime;
+use icu::datetime::{TypedDateTimeFormatter, options::length};
+
+const LOCALE: Locale = locale!("ja");
+
+include!("../my-data/mod.rs"); // defines BakedDataProvider
+
+fn main() {
+    let unstable_provider = BakedDataProvider;
+
+    let options = length::Bag::from_date_time_style(length::Date::Long, length::Time::Medium);
+
+    let dtf = TypedDateTimeFormatter::try_new_unstable(&unstable_provider, &LOCALE.into(), options.into())
+        .expect("Failed to initialize DateTimeFormatter");
+
+    let date = DateTime::try_new_gregorian_datetime(2020, 10, 14, 13, 21, 28)
+        .expect("Failed to create a datetime.");
+
+    let formatted_date = dtf.format(&date);
+
+    println!("📅: {}", formatted_date);
+}
+```
+
+With this provider, we can use the `unstable` constructors. These are only guaranteed to work if the `BakedDataProvider` was generated with the same version of ICU4X that you are building with, but if you build the data as part of your a build pipeline, that shouldn't be a problem.
+
+You can also make the `BakedDataProvider` implement the `AnyProvider` trait, so that it can be used with `_with_any_provider` constructors. Using these constructors is slightly less performant than the `unstable` ones, but, as the name suggests, stable across (minor) releases.
+
+```rust
+include!("../my-data/mod.rs");
+include!("../my-data/any.rs");
+let _any_provider = BakedDataProvider;
+```
+
+## `dir` and `FsDataProvider`
+
+The `dir` format will generate a directory tree of data files in JSON (although the `--syntax` option can be used to generate `postcard` or `bincode` files, which doesn't have many practical uses).
+
+Let's give it a try:
+
+```console
+$ icu4x-datagen --cldr-tag latest --icuexport-tag latest --out my-data --format dir --keys-for-bin target/debug/myapp --locales ja
+```
+
+This directory can be read by the `FsDataProvider` from the `icu_provider_fs` crate. You will also need to activate the feature for the chosen syntax on the `icu_provider` crate.
+
+Same as `BlobDataProvider`, this also a buffer provider, so you will need to activate `icu`'s `serde` feature and use the `with_buffer_provider` constructors.
+
+```console
+$ cargo add icu --features serde
+$ cargo add icu_provider --features deserialize_json
+$ cargo add icu_provider_fs
```

Comment:
I think this is really weird. The features should be on `icu_provider_fs`, so that clients don't have to add `icu_provider` just to set the feature

Reply:
I agree, we can try to fix this in the next release?

Reply:
I disagree with feature duplication. It is a feature of `icu_provider` because that's the thing that exports AsDeserializingBufferProvider (which admittedly most users shouldn't know or care about). It's semantically incorrect to add the feature to `icu_provider_fs` because that crate is agnostic to the chosen serialization format. I used to be in the camp that we should duplicate features, but I've drifted away from that position.

Reply:
`icu_provider` is an internal crate for all intents and purposes. Clients will never have direct dependencies on it (except for when baking tells them exactly how to add it). `icu_provider_fs` is the crate that clients will use, and that crate should be configurable directly with features.